tests/unittests/tests-analog_util/tests-analog_util.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "embUnit.h"
+#include "tests-analog_util.h"
+
+#include "analog_util.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+typedef struct {
+    int expected;
+    int sample;
+    int min;
+    int max;
+    adc_res_t res;
+} test_values_t;
+
+/* Arbitrarily chosen test vectors */
+/* TODO: Choose test vectors in a more qualified manner to catch any edge cases */
+static test_values_t test_data[] = {
```

Comment:
I was wondering what it meant to have min >= max, and by checking the API I found its forbidden:

https://github.com/RIOT-OS/RIOT/blob/2018.01/sys/include/analog_util.h#L36

Really adding an `assert` there could be better.

Reply:
I think it is better to update the docs than to add an artificial limitation like this.
Times where it can be useful to have min > max is when measuring a sensor where a higher measured voltage means a lower physical value. For example a thermistor can be connected so that the measured voltage goes down when the temperature goes up.

Reply:
I updated the docs for the function to remove the requirement on min<max

Reply:
Ok for the updated doc. It would be perfect if you could put your comment in the commit message when changing the API.

Reply:
Added a notice in the main refactor commit regarding why it can be useful.